python_modules/dagster/dagster/core/definitions/decorators/solid_decorator.py

```diff
@@ -373,6 +374,20 @@ def resolve_checked_solid_fn_inputs(
                 used_inputs.add(param.name)
 
     undeclared_inputs = explicit_names - used_inputs
+
+    if decorator_name == "@graph":
+        undeclared_nothing_inputs = [
+            inp.name
+            for inp in explicit_input_defs
+            if inp.dagster_type.kind == DagsterTypeKind.NOTHING and inp.name in undeclared_inputs
+        ]
+        if undeclared_nothing_inputs:
+            undeclared_nothing_inputs_printed = ", ".join(undeclared_nothing_inputs)
+            raise DagsterInvalidDefinitionError(
+                f"Nothing-type input definitions declared in {decorator_name} "
+                "must be specified as input arguments to the decorated function."
```

Comment:
nit: maybe a chance to add some explainer here that they will need to map the nothing to any ops they wish to have the dependency. Wording concisely may be tough but i think this is pretty much the best place to explain that to the user.